src/controllers/company.number.controller.ts

```diff
@@ -52,6 +58,45 @@ const route = async (req: Request, res: Response, next: NextFunction): Promise<v
       templateName: templatePaths.COMPANY_NUMBER,
     });
   }
+
+  const companyNumber: string = req.body.companyNumber;
+  try {
+    logger.info(`Retrieving company profile for company number ${companyNumber}`);
+    const token: string = req.chSession.accessToken() as string;
+
+    // TODO: Place this call somewhere more global so that it not called on check company as well
+    const company: PTFCompanyProfile = await getCompanyProfile(companyNumber, token);
+
+    const existing = req.chSession.data[keys.PTF_SESSION];
+    if (existing) {
+      await sessionService.updatePTFSessionValue(req.chSession,
+          keys.COMPANY_NUMBER_IN_CONTEXT, company.companyNumber);
+    } else  {
+      await sessionService.createPromiseToFileSession(req.chSession, company.companyNumber);
```

Comment:
It would be better if this was created by middleware for all routes.

Reply:
I agree but I would like to focus on this as part fo another ticket as I think there are few issues to look at with session handling so will add a TODO here.